lib/rust/ensogl/component/grid-view/src/header.rs

```diff
@@ -365,6 +366,44 @@ impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> GridView<E, HeaderEntry>
     }
 }
 
+impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> FrpNetworkProvider
+    for GridView<E, HeaderEntry>
+{
+    fn network(&self) -> &frp::Network {
+        self.model.grid.network()
+    }
+}
+
+impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> application::View
+    for GridView<E, HeaderEntry>
+{
+    fn label() -> &'static str {
+        "GridViewWithHeaders"
+    }
```

Comment:
Again, why do you need bounds in order to get the name of this component?

Reply:
As I wrote on Discord: the `label` method is a method of the `application::View` trait; the trait also has a `new` method; the proposed implementation of that `new` method needs to call `GridView<E, HeaderEntry>::new(app)`, which is only defined for the bounds specified as: `impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> GridView<E, HeaderEntry>`. Given that, this whole implementation of the `application::View` trait is constrained by those bounds, AFAIU (thus, including the `label` trait method implementation). Notably, relaxing the bounds on the trait implementation makes the compiler raise an error due to the type returned by `GridView<...>::new` not being equal to the `Self` type of the trait implementation.